Add support for fast-check@3.0.1

Currently installing `zod-fast-check` raises error:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: mypackage@0.0.1
npm ERR! Found: fast-check@3.0.1
npm ERR! node_modules/fast-check
npm ERR!   dev fast-check@"^3.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer fast-check@"^2.0.0" from zod-fast-check@0.5.2
npm ERR! node_modules/zod-fast-check
npm ERR!   dev zod-fast-check@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
```

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "zod-fast-check",
-  "version": "0.7.0",
+  "version": "0.7.1",
   "description": "Generate fast-check arbitraries from Zod schemas.",
   "main": "dist/zod-fast-check.js",
   "types": "dist/zod-fast-check.d.ts",
@@ -37,7 +37,7 @@
   "homepage": "https://github.com/DavidTimms/zod-fast-check",
   "devDependencies": {
     "@types/jest": "^28.1.6",
-    "fast-check": "^2.24.0",
+    "fast-check": "^3.0.1",
     "jest": "^26.6.3",
     "prettier": "^2.2.1",
     "ts-jest": "^26.4.4",
@@ -45,7 +45,7 @@
     "zod": "^3.19.1"
   },
   "peerDependencies": {
-    "fast-check": "^2.0.0",
+    "fast-check": "^3.0.1",
     "zod": "^3.18.0"
   }
 }

--- a/src/zod-fast-check.ts
+++ b/src/zod-fast-check.ts
@@ -241,7 +241,10 @@ const arbitraryBuilders: ArbitraryBuilders = {
 
     if (maxLength === null) maxLength = 2 * minLength + 10;
 
-    const unfiltered = fc.string(minLength, maxLength);
+    const unfiltered = fc.string({
+      minLength,
+      maxLength,
+    });
 
     if (hasRegexCheck) {
       return filterArbitraryBySchema(unfiltered, schema, path);
@@ -275,9 +278,16 @@ const arbitraryBuilders: ArbitraryBuilders = {
     }
 
     if (isInt) {
-      return fc.integer(min, max);
+      return fc.integer({ min, max });
     } else {
-      return fc.double(min, max);
+      return fc.double({
+        min,
+        max,
+        // fast-check 3 considers NaN to be a Number by default,
+        // but Zod does not consider NaN to be a Number
+        // see https://github.com/dubzzz/fast-check/blob/main/packages/fast-check/MIGRATION_2.X_TO_3.X.md#new-floating-point-arbitraries-
+        noNaN: true,
+      });
     }
   },
   ZodBigInt() {
@@ -302,11 +312,10 @@ const arbitraryBuilders: ArbitraryBuilders = {
   ) {
     const minLength = schema._def.minLength?.value ?? 0;
     const maxLength = Math.min(schema._def.maxLength?.value ?? 10, 10);
-    return fc.array(
-      recurse(schema._def.type, path + "[*]"),
+    return fc.array(recurse(schema._def.type, path + "[*]"), {
       minLength,
-      maxLength
-    );
+      maxLength,
+    });
   },
   ZodObject(
     schema: ZodObject<ZodRawShape>,
@@ -334,9 +343,13 @@ const arbitraryBuilders: ArbitraryBuilders = {
     unsupported(`Intersection`, path);
   },
   ZodTuple(schema: ZodTuple, path: string, recurse: SchemaToArbitrary) {
-    return fc.genericTuple(
-      schema._def.items.map((item, index) => recurse(item, `${path}[${index}]`))
-    ) as Arbitrary<[any, ...any[]]>;
+    // As of https://github.com/dubzzz/fast-check/commit/18afcb2d6fe9ed9b93abad795660687c87e74f61
+    // fc.genericTuple has been replaced with fc.tuple
+    return fc.tuple(
+      ...schema._def.items.map((item, index) =>
+        recurse(item, `${path}[${index}]`)
+      )
+    );
   },
   ZodRecord(schema: ZodRecord, path: string, recurse: SchemaToArbitrary) {
     return fc.dictionary(
@@ -350,8 +363,10 @@ const arbitraryBuilders: ArbitraryBuilders = {
     return fc.array(fc.tuple(key, value)).map((entries) => new Map(entries));
   },
   ZodSet(schema: ZodSet, path: string, recurse: SchemaToArbitrary) {
+    // As of https://github.com/dubzzz/fast-check/commit/8b8e3edd913ce6172339503d20cf3f67eb9676a1
+    // fc.set is now fc.uniqueArray
     return fc
-      .set(recurse(schema._def.valueType, path + ".(value)"))
+      .uniqueArray(recurse(schema._def.valueType, path + ".(value)"))
       .map((members) => new Set(members));
   },
   ZodFunction(

